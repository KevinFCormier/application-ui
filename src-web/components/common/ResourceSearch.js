/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { Search } from 'carbon-components-react'
import msgs from '../../../nls/platform.properties'

/* FIXME: Please fix disabled eslint rules when making changes to this file. */
/* eslint-disable react/prop-types */
const ResourceSearch = ({
  handleSearch,
  searchLabel,
  searchPlaceholder,
  renderSmallSearch,
  searchValue
}, context) =>
  <Search
    className='resource-search'
    small={renderSmallSearch}
    aria-label={msgs.get('search.label', context.locale)}
    labelText={searchLabel ? searchLabel : msgs.get('search.label', context.locale)}
    placeHolderText={searchPlaceholder ? searchPlaceholder : msgs.get('search.placeholder', context.locale)}
    value={searchValue}
    onChange={handleSearch} />

ResourceSearch.contextTypes = {
  locale: PropTypes.string
}

export default ResourceSearch
