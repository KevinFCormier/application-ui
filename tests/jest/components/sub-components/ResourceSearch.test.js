/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'

// mock the uuid creation in ResourceSearch
jest.mock('uuid', () => {
  return { v4: jest.fn(() => 1) }
})
import ResourceSearch from '../../../../src-web/components/common/ResourceSearch'

describe('NoResource component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <ResourceSearch
        handleSearch="handleSearch"
        searchLabel="true"
        searchPlaceholder="searchPlaceholder"
        renderSmallSearch="renderSmallSearch"
        searchValue="value"
      >
        <div className="child">Test</div>
      </ResourceSearch>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
