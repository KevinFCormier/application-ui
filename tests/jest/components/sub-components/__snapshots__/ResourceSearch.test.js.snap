// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResource component renders as expected 1`] = `
<div
  className="bx--search bx--search-with-options bx--search--sm resource-search"
  role="search"
>
  <svg
    alt="search"
    aria-label="search"
    className="bx--search-magnifier"
    fill={undefined}
    fillRule="evenodd"
    height="16"
    name="search--glyph"
    role="img"
    style={undefined}
    viewBox="0 0 16 16"
    width="16"
  >
    <title>
      search
    </title>
    <path
      d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zm10 13.8L13.8 16l-3.6-3.6 2.2-2.2z"
    />
    <path
      d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"
    />
  </svg>
  <label
    className="bx--label"
    htmlFor="resource-search"
  >
    true
  </label>
  <input
    aria-label="Search"
    className="bx--search-input"
    id="resource-search"
    onChange={[Function]}
    placeholder="searchPlaceholder"
    type="text"
    value="value"
  />
  <svg
    alt="close"
    aria-label="close"
    className="bx--search-close"
    fill={undefined}
    fillRule="evenodd"
    height="16"
    name="close--glyph"
    onClick={[Function]}
    role="img"
    style={undefined}
    viewBox="0 0 16 16"
    width="16"
  >
    <title>
      close
    </title>
    <path
      d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm3.5 10.1l-1.4 1.4L8 9.4l-2.1 2.1-1.4-1.4L6.6 8 4.5 5.9l1.4-1.4L8 6.6l2.1-2.1 1.4 1.4L9.4 8l2.1 2.1z"
    />
  </svg>
</div>
`;
